atom/browser/atom_browser_client.cc

```diff
@@ -90,14 +98,22 @@ bool AtomBrowserClient::ShouldCreateNewSiteInstance(
     content::BrowserContext* browser_context,
     content::SiteInstance* current_instance,
     const GURL& url) {
-
+  if (g_suppress_renderer_process_restart) {
+    g_suppress_renderer_process_restart = false;
+    return false;
+  }
   if (url.SchemeIs(url::kJavaScriptScheme))
     // "javacript:" scheme should always use same SiteInstance
     return false;
 
-  if (!IsRendererSandboxed(current_instance->GetProcess()->GetID()))
+  int process_id = current_instance->GetProcess()->GetID();
+  if (g_reuse_renderer_process_for_new_window) {
+    // native window.open can reuse renderer process
+    g_reuse_renderer_process_for_new_window = false;
+  } else if (!IsRendererSandboxed(process_id)) {
```

Comment:
What is the purpose of `g_reuse_renderer_process_for_new_window` variable , can't this condition be just `if(!IsRendererSandboxed(process_id) && !RendererUsesNativeWindowOpen(process_id))`  ?